### ISSUE TITLE
Remote Config fetchAndActivate does not honour minimumFetchInterval in RemoteConfigSettings

### ISSUE THREAD

**Issue by yliu342** [firebase/firebase-ios-sdk#4740]
### [REQUIRED] Step 1: Describe your environment

  * Xcode version: 11.3
  * Firebase SDK version: 6.14.0
  * Firebase Component: Remote Config
  * Component version: 6.14.0
  * Installation method: `Carthage` 

#### Steps to reproduce:
On app launch, set `minimumFetchInterval` to 10 sec,
Call `fetchAndActivate` with a completion handler.
The first time it will always fetch from the server.
Then wait for a couple of min and re launch the app
You can see from the log
`[Firebase/RemoteConfig][I-RCN000067] Successfully set configSettings. Developer Mode: false, Minimum Fetch Interval:10.000000, Fetch timeout:60.000000`
`[Firebase/RemoteConfig][I-RCN000051] Returning cached data.`
`[Firebase/RemoteConfig][I-RCN000069] Most recently fetched config is already activated.`
It does not make a call to the backend to retrieve the latest Config.

The `minimumFetchInterval` is not taking effect.
#### Relevant Code:

```
let remoteConfig = RemoteConfig.remoteConfig()
let settings = RemoteConfigSettings()
settings.minimumFetchInterval = 10 
remoteConfig.setDefaults(fromPlist: "RemoteConfigDefault")
remoteConfig.configSettings = settings
remoteConfig.fetchAndActivate(completionHandler: { (status, error) in
      ....
})
```

**Comment by yliu342** [comment]
After some digging around I found out that setting the `configSettings` is an async call. If I make a call `fetchAndActivate` right after setting the configSettings then the minimumFetchInterval has not been take affect so the Default value is always used.

Not sure if this is considered as a bug or by design? If it is by design maybe update the documentation?

**Comment by dmandar** [comment]
Thanks for reporting. It makes sense for configSettings to return a completion handler. Internal bug b/148652144 tracking this.

**Comment by toandk** [comment]
I got the same problem. Is there any workaround solution for this issue? @dmandar @yliu342

**Comment by dmandar** [comment]
As a workaround you can call fetchwithexpirationduration: to explicitly set the minimumfetchinterval for just that fetch call.

**Comment by jyothishd** [comment]
Am also experiencing the same behavior, Only the first time the remote config is fetched from the server. Even on setting minimumFetchInterval as zero, the  remote config is not fetched again from remote. It is fetched from cache only. Can this be fixed on priority.

**Comment by paulb777** [comment]
We have a fix going into Firebase 6.25.0.

I wasn't able to reproduce the issue with the `fetchwithexpirationduration:` API but I believe I saw an issue with it earlier in my testing, so there could also be a separate networking or backend issue unrelated to the SDK.  If you have a repro case for a failure with the API with the `expirationDuration` parameter, please create another issue and we'll continue to investigate.

**Comment by mlachmish** [comment]
I can confirm that the issue reproduced in Firebase 6.21.0 and has been **resolved** in Firebase 6.25.0

**Comment by jhandguy** [comment]
This issue is still happening for us with Firebase `6.26.0`.
fetchAndActivate still ignores the minimumFetchInterval value.
